lib/openfeature/rollout/provider.rb

```diff
@@ -51,8 +51,8 @@ def initialize(rollout_client: nil)
         @rollout_client = rollout_client
       end
 
-      def fetch_boolean_value(flag_key:, default_value:, evaluation_context: nil)
-        res = @rollout_client.active?(flag_key, evaluation_context&.targeting_key)
+      def fetch_boolean_value(flag_key:, default_value: nil, evaluation_context: nil)
+        res = @rollout_client.active?(flag_key, evaluation_context)
```

Comment:
I don't think that we can simplify the evaluation context from an instance of `OpenFeature::SDK::EvaluationContext` here.  The specification requires this to be at least a hash so extra values can be specified: [Requirement 3.1.2](https://openfeature.dev/specification/sections/evaluation-context#requirement-312).

Additionally, other providers, should a user choose to switch to them, rely on this existing object behavior:

The `flagd` provider relying on `#fiields` being defined (as it is in `OpenFeature::SDK::EvaluationContext`):

https://github.com/open-feature/ruby-sdk-contrib/blob/main/providers/openfeature-flagd-provider/lib/openfeature/flagd/provider/client.rb#L100

The LaunchDarkly Provider relying on calling `#field` and in addition expecting to use the behavior of requirement 3.1.2 to have a custom field:

https://github.com/launchdarkly/openfeature-ruby-server/blob/main/lib/ldclient-openfeature/impl/context_converter.rb#L28

Interactions like this become problematic if something like the meta provider is used to combine these.

I agree that it is unfortunate that OpenFeature is so much more verbose than Rollout at client call-sites.  Instead of changing the SDK in a spec-incompatible way, it might make more sense to use  Factory or Builder pattern in the application itself to wrap up the boilerplate.

Reply:
Okay, yeah, good call. I've backed this one out in [2c78a93](https://github.com/bryanalves/openfeature-rollout-provider/pull/1/commits/2c78a936842d04f19d48a570394de2074c83b088)